deepfence_worker/ingesters/common.go

```diff
@@ -97,31 +106,41 @@ func CommitFuncStatus[Status any](ts utils.Neo4jScanType) func(ns string, data [
 	}
 }
 
-// also handles status deduplication
+// handles status deduplication
+// and maintains the order of the messages
 func statusesToMaps[T any](data []T) []map[string]interface{} {
-	statusBuff := map[string]map[string]interface{}{}
+	type Entry struct {
+		Posn int
+		Data map[string]interface{}
+	}
```

Comment:
I would rather keep previous code here as this is not solving anything new from what I understand.
I am fine with the allocation optimization:
```
statuses := make([]map[string]interface{}, len(statusBuff))
```
But the `Posn` introduction is not solving anything new, is it?